src/frontend/src/table.rs

```diff
@@ -272,6 +331,46 @@ impl DistTable {
         }
         Ok(())
     }
+
+    async fn find_datanode_instances(
+        &self,
+        regions: Vec<RegionNumber>,
```

Comment:
nitpick: Could we use slice?
```suggestion
        regions: &[RegionNumber],
```